### PR TITLE
fix: move findPort as static method on Server

### PR DESCRIPTION
**Pull request by anshumanv**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
existing 

### Breaking Changes
nope

### CODE REVIEW

File: test/server/utils/findPort.test.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 const http = require('http');
 const portfinder = require('portfinder');
-const findPort = require('../../../lib/utils/findPort');
+const findPort = require('../../../lib/Server').findPort;
```
Comment (snitin315) [inline test/server/utils/findPort.test.js:5]: Ideally we should move this test to `Server.test.js`

Comment (anshumanv) [reply test/server/utils/findPort.test.js:5]: no problem here too, isolated test

Comment (alexander-akait) [reply test/server/utils/findPort.test.js:5]: Yes, move it to `Server.test.js`

File: lib/Server.js [hunk 2]
```diff
@@ -837,6 +838,32 @@ class Server {
     };
   }
 
+  static findPort(port) {
```
Comment (alexander-akait) [inline lib/Server.js:841]: Maybe getFreePort?

Comment (anshumanv) [reply lib/Server.js:841]: yep sounds good

**Comment by anshumanv** [comment]
@alexander-akait @snitin315 👍

File: lib/Server.js [hunk 3]
```diff
@@ -31,7 +33,6 @@ const getCompilerConfigArray = require('./utils/getCompilerConfigArray');
 const getStatsOption = require('./utils/getStatsOption');
 const getColorsOption = require('./utils/getColorsOption');
 const setupExitSignals = require('./utils/setupExitSignals');
-const findPort = require('./utils/findPort');
```
Comment (alexander-akait) [inline lib/Server.js:34]: Forgot one note - we should keep this file due https://github.com/webpack/webpack-cli/blob/webpack-cli%404.6.0/packages/serve/src/startDevServer.ts and old other verions with todo to remove it in the next major release and reexport static method

**Review by alexander-akait — CHANGES_REQUESTED** [review]
We should avoid break old version of CLI

File: lib/Server.js (hunk 3 above)
Comment (anshumanv) [reply lib/Server.js:34]: oh yep lets keep it

**Comment by anshumanv** [comment]
Fixed

File: lib/utils/findPort.js [hunk 4]
```diff
@@ -1,32 +1,6 @@
...
-
-  // Try to find unused port and listen on it for 3 times,
-  // if port is not specified in options.
-  const defaultPortRetry = parseInt(process.env.DEFAULT_PORT_RETRY, 10) || 3;
-
-  return pRetry(runPortFinder, { retries: defaultPortRetry });
-}
-
-module.exports = findPort;
+module.exports = getFreePort;
```
Comment (alexander-akait) [inline lib/utils/findPort.js:6]: Can we add simple tests for this and put them in `Server.test.api` too, we will remove it in future, so it is fine place

Comment (anshumanv) [reply lib/utils/findPort.js:6]: oh yep lets do it

Comment (anshumanv) [reply lib/utils/findPort.js:6]: done

**Review by alexander-akait — APPROVED** [review]

**Review by snitin315 — APPROVED** [review]

**Comment by alexander-akait** [comment]
@anshumanv Can you rebase?

**Comment by anshumanv** [comment]
Done 👍

File: test/server/Server.test.js [hunk 5]
```diff
@@ -1,8 +1,12 @@
 'use strict';
 
 const { relative, sep } = require('path');
+const http = require('http');
 const webpack = require('webpack');
 const sockjs = require('sockjs/lib/transport');
+const getFreePort = require('../../lib/Server').getFreePort;
```
Comment (snitin315) [inline test/server/Server.test.js:7]: `Server` already imported below

**Review by snitin315 — APPROVED** [review]

**Pull request merged**